### PR TITLE
Re(Build) selected packages on existing DD4hep

### PR DESCRIPTION
**Pull request by andresailer**
First prototype for feature request from @bcouturi 

Add DD4HEP_USE_EXISTING_DD4HEP option which together with DD4HEP_BUILD_PACKAGES can be used to rebuild, for example only DDG4.

Adapted DD4hepConfig to create a DD4hepSelectedConfig.cmake.
I tested that one can create a project which uses both DD4hep and DD4hepSelected

BEGINRELEASENOTES

* CMake: Add DD4HEP_USE_EXISTING_DD4HEP option which together with DD4HEP_BUILD_PACKAGES can be used to rebuild, for example only DDG4.
   * This create a new Package called "DD4hepSelected" which can then be used alongside the full DD4hep Package in a third project.

ENDRELEASENOTES

### CODE REVIEW

File: cmake/DD4hepConfig.cmake.in [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
+set ( @CMAKE_PROJECT_NAME@_SOVERSION          "@DD4hep_SOVERSION@" )
 set ( DD4HEP_USE_GEANT4         @DD4HEP_USE_GEANT4@ )
 set ( DD4HEP_USE_CLHEP          @DD4HEP_USE_CLHEP@ )
 set ( DD4HEP_USE_LCIO           @DD4HEP_USE_LCIO@ )
 set ( DD4HEP_USE_XERCESC        @DD4HEP_USE_XERCESC@ )
-set ( DD4hep_BUILD_CXX_STANDARD @CMAKE_CXX_STANDARD@ )
+set ( @CMAKE_PROJECT_NAME@_BUILD_CXX_STANDARD @CMAKE_CXX_STANDARD@ )
 
-include ( ${DD4hep_DIR}/cmake/DD4hep.cmake )
+include ( ${@CMAKE_PROJECT_NAME@_DIR}/cmake/@CMAKE_PROJECT_NAME@.cmake )
```
Comment (andresailer) [inline cmake/DD4hepConfig.cmake.in]: @bcouturi This change is probably the culprit for the error you observe:
```suggestion
include ( ${@CMAKE_PROJECT_NAME@_DIR}/cmake/DD4hep.cmake )
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -140,9 +142,31 @@ if( DEFINED DD4HEP_WITH_GEANT4 OR DEFINED DD4HEP_WITH_GEAR )
   dd4hep_print_cmake_options( OPTIONAL "Obsolete options: only the following are allowed:" ERROR 1 )
 endif()
 
+IF(DD4HEP_USE_EXISTING_DD4HEP)
+
+  MESSAGE(STATUS "|!!> Using pre-existing DD4hep: ${DD4HEP_USE_EXISTING_DD4HEP}")
+  SET(CMAKE_PROJECT_NAME DD4hep_EXTRA)
```
Comment (andresailer) [inline CMakeLists.txt]: I think ``DD4hepSelected` would be better.
```suggestion
  SET(CMAKE_PROJECT_NAME DD4hepSelected)
```

**Comment by petricm** [comment]
@bcouturi do this fulfill the LHCb requirements?

File: CMakeLists.txt (hunk 2 above)
Comment (petricm) [reply CMakeLists.txt]: I agree

File: CMakeLists.txt [hunk 3]
```diff
@@ -144,9 +146,31 @@ if( DEFINED DD4HEP_WITH_GEANT4 OR DEFINED DD4HEP_WITH_GEAR )
...
 endif()
 
+IF(DD4HEP_USE_EXISTING_DD4HEP)
+
+  MESSAGE(STATUS "|!!> Using pre-existing DD4hep: ${DD4HEP_USE_EXISTING_DD4HEP}")
+  SET(CMAKE_PROJECT_NAME DD4hepSelected)
+  dd4hep_set_version(${CMAKE_PROJECT_NAME}
+    MAJOR ${${CMAKE_PROJECT_NAME}_VERSION_MAJOR}
+    MINOR ${${CMAKE_PROJECT_NAME}_VERSION_MINOR}
+    PATCH ${${CMAKE_PROJECT_NAME}_VERSION_PATCH} )
```
Comment (petricm) [inline CMakeLists.txt]: Do we need a check that if you build for instance DDG4 against an already build DDCore that the versions match? I mean the version of the sources currently being build against the version of the binaries?

**Pull request merged**